File: messages/src/main/java/com/thundashop/core/usermanager/data/Company.java
package com.thundashop.core.usermanager.data;

import com.thundashop.core.common.DataCommon;
import java.util.ArrayList;
import java.util.List;

/**
 *
 * @author ktonder
 */
public class Company extends DataCommon implements Comparable<Company> {
    public Address invoiceAddress;
    public Address address;
    public String vatNumber = "";
    public String type = "";
    public String name = "";
    public String prefix = "47";
    public String phone = "";
    public boolean vatRegisterd = true;
    public String invoiceEmail = "";
    public String website = "";
    public String email = "";
    public String contactPerson = "";
    public String groupId = "";
    
    public String reference = "";
    public String description = "";
    
    /* This are legacy attributes, replaced with address field*/
    public String streetAddress = "";
    public String postnumber = "";
    public String country = "";
    public String city = "";
    
    public String invoiceReference = "";
    
    public String language = "";
    public String currency = "";
    
    public List<Company> subCompanies = new ArrayList();
    
    public String companyLeaderUserId = "";
    
    public boolean needConfirmation = false;

    @Override
    public int compareTo(Company o) {
        if(o != null && o.name != null && name != null) {
            return o.name.compareTo(name);
        }
        return 0;
    }
    
    
}

File: html5apps/terminal/events/core/usermanager/data/Company.php
<?php
class core_usermanager_data_Company extends core_common_DataCommon  {
	/** @var core_usermanager_data_Address */
	public $invoiceAddress;

	/** @var core_usermanager_data_Address */
	public $address;

	/** @var String */
	public $vatNumber;

	/** @var String */
	public $type;

	/** @var String */
	public $name;

	/** @var String */
	public $prefix;

	/** @var String */
	public $phone;

	/** @var String */
	public $vatRegisterd;

	/** @var String */
	public $invoiceEmail;

	/** @var String */
	public $website;

	/** @var String */
	public $email;

	/** @var String */
	public $contactPerson;

	/** @var String */
	public $groupId;

	/** @var String */
	public $reference;

	/** @var String */
	public $description;

	/** @var String */
	public $streetAddress;

	/** @var String */
	public $postnumber;

	/** @var String */
	public $country;

	/** @var String */
	public $city;

	/** @var String */
	public $invoiceReference;

	/** @var core_usermanager_data_Company[] */
	public $subCompanies;

	/** @var String */
	public $companyLeaderUserId;

	/** @var String */
	public $needConfirmation;

}
?>